integrations/chatwoot/src/setup/handler.ts

```diff
@@ -0,0 +1,146 @@
+import * as bp from '.botpress'
+import { ChatwootWebhookPayload, chatwootWebhookPayloadSchema } from '../misc/types'
+
+export const handler: bp.IntegrationProps['handler'] = async ({ req, client }) => {
+  const payload = chatwootWebhookPayloadSchema.parse(JSON.parse(req.body || '{}'))
+
+  if (payload.event === 'conversation_status_changed' && payload.status === 'resolved' && payload.id) {
+    await handleConversationResolvedById(payload.id.toString(), client)
+    return
+  }
+
+  if (payload.event !== 'message_created') return
+
+  // message_type: 0/"incoming" or 1/"outgoing" - string for API calls, number for webhooks
+  const isOutgoing = payload.message_type === 1 || payload.message_type === 'outgoing'
+  const isIncoming = payload.message_type === 0 || payload.message_type === 'incoming'
+
+  if (isOutgoing) {
+    const chatwootConvId = payload.conversation?.id?.toString()
+    if (!chatwootConvId) return
+
+    const { conversations } = await client.listConversations({
+      tags: { id: chatwootConvId },
+    })
+
+    const hitlConv = conversations.find((c) => c.tags?.bpUserId)
+    if (!hitlConv) {
+      return
```

Comment:
todo: please either leave a comment explaining why it was not found, a log or an error. If I understand this correctly, this is a human agent message not being handled by botpress in hitl mode. It seems like we would need to do something if there is a failure.